.github/workflows/circleci-trigger.yml

```diff
@@ -0,0 +1,89 @@
+name: CircleCI tests trigger
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches:
+      - main
+
+env:
+  PYTHONUNBUFFERED: 1
+  FORCE_COLOR: 1
+
+jobs:
+  circleci-trigger:
+    name: CircleCI tests trigger
+    environment: circleci_secrets
+    runs-on: ubuntu-latest
+    steps:
+      - name: Ping CircleCI pipeline
+        run: |
+          if [[ "${{ github.ref_type }}" == "tag" ]]; then
+            github_tag=${{ github.ref_name }}
+          else github_tag=""
+          fi
+
+          CREATED_PIPELINE=$(curl --request POST \
+            --url "${{ secrets.QA_PIPELINE_URL }}" \
+            --header "Circle-Token: ${{ secrets.CIRCLECI_API_KEY }}" \
+            --header "content-type: application/json" \
+            --data '{"branch":"${{ github.head_ref }}", "parameters":{"branch":"${{ github.head_ref }}", "vizro_branch": "${{ github.head_ref }}", "tag": "$github_tag"}}' \
+          | jq -r '.id'
+          )
+          if [[ "$CREATED_PIPELINE" == "null" ]]; then
+            CREATED_PIPELINE=$(curl --request POST \
+            --url "${{ secrets.QA_PIPELINE_URL }}" \
+            --header "Circle-Token: ${{ secrets.CIRCLECI_API_KEY }}" \
+            --header "content-type: application/json" \
+            --data '{"branch":"main", "parameters":{"branch":"main", "vizro_branch": "${{ github.head_ref }}", "tag": "$github_tag"}}' \
+            | jq -r '.id'
+            )
+          fi
+          echo "my created pipeline"
+          echo $CREATED_PIPELINE
+
+          sleep 60
+
+          created_workflow_status=$(curl --request GET \
+              --url "https://circleci.com/api/v2/pipeline/$CREATED_PIPELINE/workflow" \
+              --header "Circle-Token: ${{ secrets.CIRCLECI_API_KEY }}" \
+              --header "content-type: application/json" \
+            | jq -r '.items[0].status'
+          )
+
+          echo $created_workflow_status
+
+          if [[ "$created_workflow_status" == "running" ]]; then
+            created_workflow_status_updated=$(curl --request GET \
+                  --url "https://circleci.com/api/v2/pipeline/$CREATED_PIPELINE/workflow" \
+                  --header "Circle-Token: ${{ secrets.CIRCLECI_API_KEY }}" \
+                  --header "content-type: application/json" \
+                | jq -r '.items[0].status'
+              )
+            while [[ "$created_workflow_status_updated" == "running" ]]
+            do
+              sleep 15
+              created_workflow_status_updated=$(curl --request GET \
+                  --url "https://circleci.com/api/v2/pipeline/$CREATED_PIPELINE/workflow" \
+                  --header "Circle-Token: ${{ secrets.CIRCLECI_API_KEY }}" \
+                  --header "content-type: application/json" \
+                | jq -r '.items[0].status'
+              )
+              echo $created_workflow_status_updated
+            done
```

Comment:
Is it possible to refactor this somehow to be less repetitive? Currently it looks a lot like:

```
created_workflow_status = do_something()

if [[ "$created_workflow_status" == "running" ]]; then
  created_workflow_status_updated = do_something()
  while [[ "$created_workflow_status_updated" == "running" ]]
  do
    sleep 15
    created_workflow_status_update = do_something()
  done
```

Can't we just do as something like this? As rewritten quickly by ChatGPT plus me.

```
get_workflow_status() {
  curl --silent --request GET \
       --url "https://circleci.com/api/v2/pipeline/$CREATED_PIPELINE/workflow" \
       --header "Circle-Token: ${{ secrets.CIRCLECI_API_KEY }}" \
       --header "content-type: application/json" \
    | jq -r '.items[0].status'
}

while created_workflow_status=$(get_workflow_status); [[ "$created_workflow_status" == "running" ]]; do
  echo $created_workflow_status
  sleep 15
done
```

Reply:
Done with it. Nice one!